EIPS/eip-5827.md

```diff
@@ -0,0 +1,86 @@
+---
+eip: 5827
+title: Auto-renewable allowance extension
+description: Extension to enable automatic renewals on allowance approvals
+author: zlace (@zlace0x), zhongfu (@zhongfu), edison0xyz (@edison0xyz)
+discussions-to: https://ethereum-magicians.org/t/eip-5827-auto-renewable-allowance-extension/10392
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-10-22
+requires: 20, 165
+---
+
+## Abstract
+
+This extension adds a recovery mechanism to [EIP-20](./eip-20.md) allowances, in which a `recoveryRate` defines the amount of token per second that the allowance recovers towards the initial maximum approval `amount`.
+
+## Motivation
+
+Currently, EIP-20 supports allowances, with which token owners can allow a spender to spend a certain amount of tokens on their behalf. However, this is not ideal in circumstances involving recurring payments (e.g. subscriptions, salaries, recurring DCA purchases).
+
+Many existing dApps circumvent this limitation by requesting that users grant a large or unlimited allowance. This presents a security risk as malicious dApps can drain users' accounts up to the allowance granted, and users may not be aware of the implications of giving an allowances.
+
+An auto-renewable allowance enables many traditional financial concepts like credit and debit limits. An account owner can specify a spending limit, and limit the amount charged to the account based on an allowance that recovers over time.
+
+
+## **Specification**
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+
+interface IERC5827 {
```

Comment:
```suggestion
interface IERC5827 is ERC20 {
```

Reply:
On the other hand, will like to keep this interface from being strictly ERC20. One possible way of implementation is via a proxy contract that existing ERC20 tokens can delegate allowance enforcement to.